wgpu-core/src/command/bundle.rs

```diff
@@ -197,14 +197,20 @@ impl RenderBundle {
                 RenderCommand::SetPipeline(pipeline_id) => {
                     let pipeline = pipeline_guard.get(pipeline_id).unwrap();
                     cmd_buf.bind_graphics_pipeline(&pipeline.raw);
-                    index_type = conv::map_index_format(pipeline.index_format);
+
+                    if let Some(index_format) = pipeline.index_format {
+                        index_type = conv::map_index_format(index_format);
+                    }
+
                     pipeline_layout_id = Some(pipeline.layout_id.value);
                 }
                 RenderCommand::SetIndexBuffer {
                     buffer_id,
+                    index_format,
                     offset,
                     size,
                 } => {
+                    index_type = conv::map_index_format(index_format);
```

Comment:
we also need to call `state.index.update_limit();` again if this format is changed here

Reply:
`execute` in `RenderBundle` doesn't really have a state, should it?